src/components/React/VolumeRenderer.js

```diff
@@ -79,23 +72,55 @@ export default connect(mapStateToProps)
             cursor-listener 
           /> 
 
-        
-          <Entity 
-            id="volumeCube" 
-            class="clickableMesh"   
-            myloader={{
-              volumeData:this.props.volumeData,rayCollided:false,
-              transferFunction:this.props.transferFunction,colorMap:this.props.colorMap,
-              opacity1:this.props.opacity1,opacity2:this.props.opacity2,
-              lowNode:this.props.lowNode,highNode:this.props.highNode,
-              alphaXDataArray:this.props.alphaXDataArray,
-              alphaYDataArray:this.props.alphaYDataArray,
-              colorMapping:this.props.colorMapping,
-              channel:this.props.channel,
-              cameraState:this.props.cameraState
-            }}   
-            position="0 0 0"
-          />
+          <VolumeConsumer>
+            {({state}) => (
+              <Entity 
+                id="volumeCube" 
+                class="clickableMesh"   
+                myloader={{
+                  volumeData: state.path,
+                  rayCollided: false,
+                  transferFunction: this.props.transferFunction,
+                  colorMap: this.props.colorMap,
+                  opacity1: this.props.opacity1,
+                  opacity2: this.props.opacity2,
+                  lowNode: this.props.lowNode, 
+                  highNode:this.props.highNode,
+                  alphaXDataArray: this.props.alphaXDataArray,
+                  alphaYDataArray: this.props.alphaYDataArray,
+                  colorMapping: this.props.colorMapping,
+                  channel: this.props.channel,
+                  cameraState: this.props.cameraState
+                }}   
+                position="0 0 0"
+              />
+            )}
```

Comment:
This is interesting to me. Can you help me understand how this works? It looks like the child to the `VolumeConsumer` is a function that generates an `Entity` component? Is `Entity` a class based component? Is that why you're using the `VolumeConsumer` here?

Reply:
If not, I think I'd encourage you to have a `useVolumeContext()` hook inside `Entity` that sets `volumeData` if `volumeData` isn't set already

Reply:
`VolumeRenderer.js` is a class based component so it wouldn't let me use `useVolumeContext()`. From what I read online the `VolumeConsumer` is how you use context when rendering class based components - it's a custom `VolumeContext.Consumer` in the same way `VolumeProvider` is a custom `VolumeContext.Provider`. Not sure about `Entity` itself, it comes from `aframe-react`

Refactoring `VolumeRenderer.js` to utilize hooks is definitely in the pipeline. Only issue is that it uses so much redux stuff it's basically a PR on its own. The whole project is (slowly) being refactored. I'll definitely switch to `useVolumeContext()` then!